### PR TITLE
Issue 223

### PR DESCRIPTION
**Pull request by ndellingwood**
Work towards disabling deprecated code.

### CODE REVIEW

**Comment by ndellingwood** [comment]
See issue #223 #219

**Comment by ndellingwood** [comment]
This PR can either be a work in progress or tested as is with more fixes in separate PR(s). So far this is just simple search/replace with sed changes. 

Remaining fixes needed:
These tests need to use `Kokkos::initialize()` and have the execution space `initialize()`/`finalize()` calls removed:
`perf_test`
```
graph/KokkosGraph_color.cpp
graph/KokkosGraph_color_d2.cpp
graph/KokkosGraph_triangle.cpp
sparse/KokkosSparse_pcg.cpp
sparse/KokkosSparse_spgemm.cpp
test_mv.cpp
```

**Comment by ndellingwood** [comment]
PR #228 contains changes to the perf_test tests which make use of deprecated features, i.e. execution space calls to initialize and finalize.

**Comment by ndellingwood** [comment]
Passes spot-check on kokkos-dev:

```
Running on machine: sems
Going to test compilers:  gcc/5.3.0 intel/16.0.1 clang/3.9.0 cuda/8.0.44
Testing compiler gcc/5.3.0
Testing compiler intel/16.0.1
  Starting job gcc-5.3.0-Serial-release
  Starting job gcc-5.3.0-Serial-hwloc-release
  PASSED gcc-5.3.0-Serial-hwloc-release
  Starting job intel-16.0.1-OpenMP-release
  PASSED gcc-5.3.0-Serial-release
Testing compiler clang/3.9.0
  Starting job intel-16.0.1-OpenMP-hwloc-release
  PASSED intel-16.0.1-OpenMP-release
  Starting job clang-3.9.0-Pthread_Serial-release
  PASSED intel-16.0.1-OpenMP-hwloc-release
Testing compiler cuda/8.0.44
  Starting job clang-3.9.0-Pthread_Serial-hwloc-release
  PASSED clang-3.9.0-Pthread_Serial-hwloc-release
  PASSED clang-3.9.0-Pthread_Serial-release
  Starting job cuda-8.0.44-Cuda_OpenMP-release
  PASSED cuda-8.0.44-Cuda_OpenMP-release
#######################################################
PASSED TESTS
#######################################################
clang-3.9.0-Pthread_Serial-hwloc-release build_time=342 run_time=1379
clang-3.9.0-Pthread_Serial-release build_time=385 run_time=3264
cuda-8.0.44-Cuda_OpenMP-release build_time=1007 run_time=1161
gcc-5.3.0-Serial-hwloc-release build_time=337 run_time=869
gcc-5.3.0-Serial-release build_time=335 run_time=1128
intel-16.0.1-OpenMP-hwloc-release build_time=387 run_time=457
intel-16.0.1-OpenMP-release build_time=438 run_time=417
#######################################################
```

File: example/fenl/KokkosKernels_ETIMacros.hpp [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 #else
-  #ifdef KOKKOS_HAVE_OPENMP
+  #ifdef KOKKOS_ENABLE_OPENMP
     #define KOKKOSKERNELS_ETI_MANGLING_TYPEDEFS()  \
                         typedef Kokkos::Device<Kokkos::OpenMP, Kokkos::OpenMP::memory_space> Kokkos_Device0Kokkos_OpenMP_Kokkos_HostSpace0; \
         typedef Kokkos::complex<double> Kokkos_complex0double0; \
         typedef long long longlong;
   #else
-    #ifdef KOKKOS_HAVE_PTHREAD
+    #ifdef KOKKOS_ENABLE_PTHREAD
```
Comment (crtrott) [inline example/fenl/KokkosKernels_ETIMacros.hpp]: I think this needs to be KOKKOS_ENABLE_THREADS?

Comment (crtrott) [reply example/fenl/KokkosKernels_ETIMacros.hpp]: Yeah I checked this needs to KOKKOS_ENABLE_THREADS

Comment (ndellingwood) [reply example/fenl/KokkosKernels_ETIMacros.hpp]: Fixed all occurrences and pushed.

**Comment by ndellingwood** [comment]
Updated PR to fix macro for THREADS backend.

**Comment by ndellingwood** [comment]
Fixed code broken by deprecated code:
`hardware_thread_id` changed to `impl_hardware_thread_id` in src/sparse/impl/KokkosSparse_spgemm_impl_speed.hpp

**Comment by ndellingwood** [comment]
Additional fixes pushed to remove `const` qualifiers from team policies that must be modifiable to use `set` methods.

**Comment by ndellingwood** [comment]
Merged in develop branch and perf_test changes from #228 (fixed conflicts).

**Comment by ndellingwood** [comment]
Spot-check passes on kokkos-dev (other issues need to be resolved before testing on apollo):

```
Running on machine: sems
Going to test compilers:  gcc/5.3.0 intel/16.0.1 clang/3.9.0 cuda/8.0.44
Testing compiler gcc/5.3.0
Testing compiler intel/16.0.1
  Starting job gcc-5.3.0-Serial-release
  Starting job gcc-5.3.0-Serial-hwloc-release
  PASSED gcc-5.3.0-Serial-release
  Starting job intel-16.0.1-OpenMP-release
  PASSED gcc-5.3.0-Serial-hwloc-release
Testing compiler clang/3.9.0
  Starting job intel-16.0.1-OpenMP-hwloc-release
  PASSED intel-16.0.1-OpenMP-release
  Starting job clang-3.9.0-Pthread_Serial-release
  PASSED intel-16.0.1-OpenMP-hwloc-release
Testing compiler cuda/8.0.44
  Starting job clang-3.9.0-Pthread_Serial-hwloc-release
  PASSED clang-3.9.0-Pthread_Serial-hwloc-release
  PASSED clang-3.9.0-Pthread_Serial-release
  Starting job cuda-8.0.44-Cuda_OpenMP-release
  PASSED cuda-8.0.44-Cuda_OpenMP-release
#######################################################
PASSED TESTS
#######################################################
clang-3.9.0-Pthread_Serial-hwloc-release build_time=394 run_time=1309
clang-3.9.0-Pthread_Serial-release build_time=451 run_time=2775
cuda-8.0.44-Cuda_OpenMP-release build_time=775 run_time=1243
gcc-5.3.0-Serial-hwloc-release build_time=334 run_time=1127
gcc-5.3.0-Serial-release build_time=330 run_time=971
intel-16.0.1-OpenMP-hwloc-release build_time=475 run_time=468
intel-16.0.1-OpenMP-release build_time=516 run_time=415
#######################################################
```

**Pull request merged**